sapl/ip_restriction_middleware.py

```diff
@@ -0,0 +1,33 @@
+from django.http import HttpResponseForbidden
+
+
+class IpRestrictionMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        # lista de IPs permitidos (localhost, redes locais, etc)
+        # https://en.wikipedia.org/wiki/Reserved_IP_addresses
+        allowed_ips = [
+            '127.0.0.1',
+            '::1',
+            '10.0.0.0/8',
+            '172.16.0.0/12',
+            '192.168.0.0/16',
+            'fc00::/7',
+            '::1',
+            'fe80::/10',
+            '192.0.2.0/24',
+            '2001:db8::/32',
+            '224.0.0.0/4',
+            'ff00::/8'
+        ]
+        # IP do cliente
+        client_ip = request.META.get('REMOTE_ADDR')
+
+        # bloqueia acesso a /metrics de IPs nao locais
+        if request.path == '/metrics':
+            if client_ip not in allowed_ips:
+                return HttpResponseForbidden('Acesso proibido')
```

Comment:
Neste caso, é uma boa prática fazer o merge dos dois `ifs`. Também adicionar log pra termos o controle de quem tá tentando 

```suggestion
        if request.path == '/metrics' and client_ip not in ALLOWED_IPS:
                logger.info(f"Tentativa de acesso a /metrics, IP: {client_ip}")
                return HttpResponseForbidden('Acesso proibido')
```